src/Engine/TesseractEngine.php

```diff
@@ -28,25 +32,43 @@ public function getText(string $imageUrl, ?array $langs = null): string
     {
         // Check the URL and fetch the image data.
         $this->checkImageUrl($imageUrl);
+
+        // Validate the languages
+        $this->validateLangs($langs);
+
         $imageResponse = $this->httpClient->request('GET', $imageUrl);
         try {
             $imageContent = $imageResponse->getContent();
         } catch (ClientException $exception) {
             throw new OcrException('image-retrieval-failed', [$exception->getMessage()]);
         }
-
-        // Run OCR.
-        $ocr = new TesseractOCR();
-        $ocr->imageData($imageContent, $imageResponse->getHeaders()['content-length'][0]);
+        $this->ocr->imageData($imageContent, $imageResponse->getHeaders()['content-length'][0]);
         if ($langs && count($langs) > 0) {
-            $ocr->lang(...$langs);
+            $this->ocr->lang(...$langs);
         }
         // Env vars are passed through by the thiagoalessio/tesseract_ocr package to the tesseract command,
         // but when they're loaded from Symfony's .env they aren't actually available (by design),
         // so we have to load this one manually. We only process one image at a time, so don't benefit from
         // multiple threads. See https://github.com/tesseract-ocr/tesseract/issues/898 for some more info.
         putenv('OMP_THREAD_LIMIT=1');
-        $text = $ocr->run();
+        $text = $this->ocr->run();
         return $text;
     }
+
+     /**
+     * @param string[]|null $langs
+     * @throws OcrException
+     */
+    public function validateLangs(?array $langs): void
+    {
+        $validLanguages = $this->ocr->availableLanguages();
+
+        if ($langs) {
+            foreach ($langs as $lang) {
+                if (!in_array($lang, $validLanguages)) {
+                    throw new OcrException('langs-param-error', []);
+                }
+            }
+        }
```

Comment:
Should be able to simplify this to something like (let's assume each engine has a `getValidLangs()` method, see my other comments):
```
$invalidLangs = array_diff($langs, $this->getValidLangs());
if (count($invalidLangs)) {
    throw new OcrException('langs-param-error', $invalidLangs);
}
```

Reply:
Good point. Thank you!